app/controllers/credentials_controller.rb

```diff
@@ -35,10 +35,18 @@ class CredentialsController < ApplicationController
   # This method requires a PUT request. The new password is in the request body.
   def update_password
     password = request.body.read
-    
-    @role.credentials.password = password
-    @role.credentials.save
-    head 204
+
+    begin
+      @role.credentials.password = password
+      @role.credentials.save(raise_on_save_failure: true)
+      head 204
+    rescue
+      # render json with unprocessable_entity only for insufficient_password_complexity otherwise exception is thrown up
```

Comment:
Suggestion, change "thrown up" to "otherwise, throw an exception"

Reply:
I don't see any "rerase" after we rescue from catch all, are we sure that we are not swallowing other errors here?
 
e.g: https://stackoverflow.com/questions/23770924/reraise-same-exception-after-catching-an-exception-in-ruby